msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/flw/TestCase2495140.java

```diff
@@ -0,0 +1,212 @@
+
+import com.microsoft.identity.client.AcquireTokenSilentParameters;
+import com.microsoft.identity.client.IAuthenticationResult;
+import com.microsoft.identity.client.ICurrentAccountResult;
+import com.microsoft.identity.client.ISingleAccountPublicClientApplication;
+import com.microsoft.identity.client.PublicClientApplication;
+import com.microsoft.identity.client.SignInParameters;
+import com.microsoft.identity.client.SilentAuthenticationCallback;
+import com.microsoft.identity.client.SingleAccountPublicClientApplication;
+import com.microsoft.identity.client.exception.MsalException;
+import com.microsoft.identity.client.msal.automationapp.R;
+import com.microsoft.identity.client.msal.automationapp.testpass.broker.AbstractMsalBrokerTest;
+import com.microsoft.identity.client.ui.automation.TestContext;
+import com.microsoft.identity.client.ui.automation.TokenRequestLatch;
+import com.microsoft.identity.client.ui.automation.TokenRequestTimeout;
+import com.microsoft.identity.client.ui.automation.annotations.SupportedBrokers;
+import com.microsoft.identity.client.ui.automation.app.AzureSampleApp;
+import com.microsoft.identity.client.ui.automation.broker.BrokerHost;
+import com.microsoft.identity.client.ui.automation.broker.BrokerMicrosoftAuthenticator;
+import com.microsoft.identity.client.ui.automation.interaction.PromptHandlerParameters;
+import com.microsoft.identity.client.ui.automation.interaction.PromptParameter;
+import com.microsoft.identity.client.ui.automation.interaction.microsoftsts.AadPromptHandler;
+import com.microsoft.identity.client.ui.automation.logging.Logger;
+import com.microsoft.identity.common.internal.util.StringUtil;
+import com.microsoft.identity.common.java.util.ThreadUtils;
+import com.microsoft.identity.labapi.utilities.client.ILabAccount;
+import com.microsoft.identity.labapi.utilities.client.LabQuery;
+import com.microsoft.identity.labapi.utilities.constants.TempUserType;
+import com.microsoft.identity.labapi.utilities.constants.UserRole;
+import com.microsoft.identity.labapi.utilities.constants.UserType;
+import com.microsoft.identity.labapi.utilities.exception.LabApiException;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+// End My Shift - In Shared device mode, global sign out should wait/cancel existing silent requests
+// and clean all data.
+// https://identitydivision.visualstudio.com/DevEx/_workitems/edit/2495140
+@SupportedBrokers(brokers = {BrokerMicrosoftAuthenticator.class, BrokerHost.class})
+public class TestCase2495140 extends AbstractMsalBrokerTest {
+    final String TAG = TestCase2495140.class.getSimpleName();
+
+    @Test
+    public void test_2495140() throws MsalException, InterruptedException, LabApiException {
+        final String username1 = mLabAccount.getUsername();
+        final String password1 = mLabAccount.getPassword();
+        Logger.i(TAG, "Performing Shared Device Registration.");
```

Comment:
Do we need logging inside UI automation test?

Reply:
It just made it easier for me to look and find logs in Android Studio, by a certain Tag